File: app/src/main/java/com/github/rooneyandshadowss/lightbulb/pickersdemo/activity/Router.java
package com.github.rooneyandshadowss.lightbulb.pickersdemo.activity;

public class Router {
    private static Router instance = null;
    private AppRouter router;

    public static synchronized Router getInstance() {
        if (null == instance) {
            instance = new Router();
        }
        return instance;
    }

    public AppRouter getRouter() {
        return router;
    }

    public void setRouter(AppRouter router) {
        this.router = router;
    }
}

File: app/src/main/java/com/github/rooneyandshadowss/lightbulb/pickersdemo/activity/Router.kt
package com.github.RooneyAndShadows.lightbulb.pickersdemo.activity

class Router {
    var router: AppRouter? = null

    companion object {
        @get:Synchronized
        var instance: Router? = null
            get() {
                if (null == field) {
                    field = Router()
                }
                return field
            }
            private set
    }
}